cmd/clusterctl/client/cluster/installer.go

```diff
@@ -90,6 +109,58 @@ func installComponentsAndUpdateInventory(components repository.Components, provi
 	return providerInventory.Create(inventoryObject)
 }
 
+// waitForReady waits till the installed components are ready.
+func (i *providerInstaller) waitForReady() error {
+	log := logf.Log
+	log.Info("Waiting for providers to be available...")
+
+	if err := i.waitDeploymentsReady(); err != nil {
+		return err
+	}
+
+	// check for services too?
+
+	return nil
+}
+
+// waitDeploymentsReady waits till the installed deployments are ready.
+func (i *providerInstaller) waitDeploymentsReady() error {
+	for _, components := range i.installQueue {
+		for _, obj := range components.Objs() {
+			if obj.GroupVersionKind().Kind == "Deployment" {
+				if err := i.waitDeploymentReady(obj); err != nil {
```

Comment:
Not sure if the optimisation is worth, but what about doing this in parallel (makes sense if this provides significant reduction in the overall wait time)

Reply:
I think it the reduction will be < 1-2 sec. We have to wait until all deployments are ready anyway and checking them shouldn't take long.

Reply:
I thought about doing parallelizing it didnt do it because of the following reasons:
* We anyway have to wait if at least one deployment is still pending.
* If you think about it while processing this sequentially, while you are waiting for one deployment you are technically also waiting for the other deployments (waiting to check status).
  * If the other deployments are ready before we check for them the very first call we would make to check the status would return "Ready" and we can continue immediately (to Stefan's point).
* With the above 2 points having the wait be sequential will also make the code more readable without any significant loss in time.

Reply:
Thanks for context!